### PR TITLE
`azurerm_nginx_deployment` dropped validation of SKU property.

### PR DESCRIPTION
**Pull request by agazeley**
## Community Note
* Please vote on this PR by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original PR to help the community and maintainers prioritize for review
* Please do not leave "+1" or "me too" comments, they generate extra noise for PR followers and do not help prioritize for review

## Description

There is a requirement from engineering to have this be a plain string field to support test SKUs that are sensitive and should not be added to a enumerated list of SKUs in the provider. The documented supported SKUs are still the publically available SKUs so not docs change is needed.

## PR Checklist

- [x] I have followed the guidelines in our [Contributing Documentation](../blob/main/contributing/README.md).
- [x] I have checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change.
- [x] I have checked if my changes close any open issues. If so please include appropriate [closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) below.
- [x] I have updated/added Documentation as required written in a helpful and kind way to assist users that may be unfamiliar with the resource / data source.
- [x] I have used a meaningful PR title to help maintainers and other users understand this change and help prevent duplicate work. 
For example: “`resource_name_here` - description of change e.g. adding property `new_property_name_here`”

## Changes to existing Resource / Data Source

- [x] I have added an explanation of what my changes do and why I'd like you to include them (This may be covered by linking to an issue above, but may benefit from additional explanation).
- [x] I have written new tests for my resource or datasource changes & updated any relevant documentation.
- [x] I have successfully run tests with my changes locally. If not, please provide details on testing challenges that prevented you running the tests.

## Testing 

- [x] My submission includes Test coverage as described in the [Contribution Guide](../blob/main/contributing/topics/guide-new-resource.md) and the tests pass. (if this is not possible for any reason, please include details of why you did or could not add test coverage)

## Change Log

Below please provide what should go into the changelog (if anything) conforming to the [Changelog Format documented here](../blob/main/contributing/topics/maintainer-changelog.md).

No changelog message is desired as we do not change support merely relaxed validation.

This is a (please select all that apply):

- [ ] Bug Fix
- [ ] New Feature (ie adding a service, resource, or data source)
- [x] Enhancement
- [ ] Breaking Change

### CODE REVIEW

**Comment by puneetsarna** [comment]
Do we drop these values from docs as well? Probably should leave them in as it helps the users.

**Comment by agazeley** [comment]
> Do we drop these values from docs as well? Probably should leave them in as it helps the users.
> 
> Other than that, this looks good. 👍

Yeah I think we should leave them in for now too. Though it is always preferred to go straight to our billing docs.

**Review by manicminer — COMMENTED** [review]
Hi @agazeley, thanks for the PR. It's a little pedantic, but could we update the docs _slightly_ to read:

```markdown
* sku - (Required) Specifies the Nginx Deployment SKU. Possible values include `publicpreview_Monthly_gmz7xq9ge3py` and `standard_Monthly`. Changing this forces a new resource to be created.
```

Thanks!

**Review by manicminer — APPROVED** [review]
Thanks @agazeley, LGTM! 👍

File: website/docs/d/nginx_deployment.html.markdown [hunk 1]
```diff
@@ -63,7 +63,7 @@ In addition to the Arguments listed above - the following Attributes are exporte
 
 * `nginx_version` - NGINX version of the Nginx Deployment.
 
-* `sku` - Name of the SKU for this Nginx Deployment.
+* `sku` - (Required) Specifies the Nginx Deployment SKU. Possible values include `standard_Monthly`. Changing this forces a new resource to be created.
```
Comment (puneetsarna) [inline website/docs/d/nginx_deployment.html.markdown:66]: this is a data source. this should be worded as a read field and not a settable field like the resource.

Comment (manicminer) [reply website/docs/d/nginx_deployment.html.markdown:66]: Oops I missed this, thanks @puneetsarna

Comment (puneetsarna) [reply website/docs/d/nginx_deployment.html.markdown:66]: no worries @manicminer and thank you for the quick fix 👍

**Pull request merged**